txnsync/mainloop.go

```diff
@@ -369,6 +369,9 @@ func (s *syncState) getPeers() (result []*Peer) {
 	peersInfo := s.node.GetPeers()
 	updatedNetworkPeers := []interface{}{}
 	updatedNetworkPeersSync := []*Peer{}
+
+	var averageDataExchangeRate uint64
```

Comment:
Just wanted to initiate a thought experiment: Is average the right metric? How would using average compare to using the maximum instead?

Reply:
If we have 4 connected relays. The connection to one of them is slow. We still want to try and build the proposal so it would get on time to most of them. So, we can't take the fastest. But taking the slowest opens to an attack vector, since it gives a slow relay the ability to control the proposal size.

Taking the average seems to be a reasonable compromise. It's definitely not perfect, but it would become better once you're done with the proposal breakout.